tests/template_test.go

```diff
@@ -0,0 +1,119 @@
+	"github.com/pulumi/pulumi/pkg/workspace"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestTemplates(t *testing.T) {
+	awsRegion := os.Getenv("AWS_REGION")
+	if awsRegion == "" {
+		awsRegion = "us-west-1"
+		fmt.Println("Defaulting AWS_REGION to 'us-west-1'.  You can override using the AWS_REGION environment variable")
+	}
+	//azureEnviron := os.Getenv("ARM_ENVIRONMENT")
+	//if azureEnviron == "" {
+	//	azureEnviron = "public"
+	//	fmt.Println("Defaulting ARM_ENVIRONMENT to 'public'.  You can override using the ARM_ENVIRONMENT variable")
+	//}
+	//azureLocation := os.Getenv("ARM_LOCATION")
+	//if azureLocation == "" {
+	//	azureLocation = "westus"
+	//	fmt.Println("Defaulting ARM_LOCATION to 'westus'.  You can override using the ARM_LOCATION variable")
+	//}
+	//gcpProject := os.Getenv("GOOGLE_PROJECT")
+	//if gcpProject == "" {
+	//	gcpProject = "pulumi-ci-gcp-provider"
+	//	fmt.Println("Defaulting GOOGLE_PROJECT to 'pulumi-ci-gcp-provider'." +
+	//		"You can override using the GOOGLE_PROJECT variable")
+	//}
+	//gcpRegion := os.Getenv("GOOGLE_REGION")
+	//if gcpRegion == "" {
+	//	gcpRegion = "us-central1"
+	//	fmt.Println("Defaulting GOOGLE_REGION to 'us-central1'.  You can override using the GOOGLE_REGION variable")
+	//}
+	//gcpZone := os.Getenv("GOOGLE_ZONE")
+	//if gcpZone == "" {
+	//	gcpZone = "us-central1-a"
+	//	fmt.Println("Defaulting GOOGLE_ZONE to 'us-central1-a'.  You can override using the GOOGLE_ZONE variable")
+	//}
+	overrides, err := integration.DecodeMapString(os.Getenv("PULUMI_TEST_NODE_OVERRIDES"))
+	if !assert.NoError(t, err, "expected valid override map: %v", err) {
+		return
+	}
+
+	base := integration.ProgramTestOptions{
+		Tracing:              "https://tracing.pulumi-engineering.com/collector/api/v1/spans",
+		ExpectRefreshChanges: true,
+		Overrides:            overrides,
+		Quick:                true,
+		SkipRefresh:          true,
+		NoParallel:           true, // we mark tests as Parallel manually when instantiating
+	}
+
+	// Retrieve the template repo.
+	repo, err := workspace.RetrieveTemplates("", false /*offline*/, workspace.TemplateKindPulumiProject)
+	assert.NoError(t, err)
+	defer assert.NoError(t, repo.Delete())
+
+	// List the templates from the repo.
+	templates, err := repo.Templates()
+	assert.NoError(t, err)
+
+	for _, template := range templates {
+		templateName := template.Name
+		t.Run(templateName, func(t *testing.T) {
+			t.Parallel()
+			if !strings.Contains(templateName, "aws") || strings.Contains(templateName, "go") {
+				t.Skipf("Skipping template test %s", templateName)
+				return
+			}
+
+			e := ptesting.NewEnvironment(t)
+			defer deleteIfNotFailed(e)
+
+			stdOut, stdErr := e.RunCommand("pulumi", "new", templateName, "-f", "-s", "template-test")
+
+			path, err := workspace.DetectProjectPathFrom(e.RootPath)
+			assert.NoError(t, err)
+			assert.NotEmpty(t, path)
+
+			if path == "" {
+				panic(fmt.Sprintf("path is empty: %v: %v", stdOut, stdErr))
+			}
```

Comment:
Why the `if path == ""` check? Aren't we asserting `path` isn't empty on line 87?

Reply:
great catch! this was debugging